src/gcodeExport.cpp

```diff
@@ -246,10 +249,6 @@ void GCodeExport::setFlavor(EGCodeFlavor flavor)
     {
         firmware_retract = true;
     }
-    else 
-    {
-        firmware_retract = false;
-    }
```

Comment:
I suggest you also remove the conditions which make `firmware_retract` true in the lines above.
If a firmware is volumetric then that should be set accordingly in the machine definition and not be concluded from other parameters.

BFB is probably an exception, because retraction is handled in a really weird way.

Given that the volumetric condition is separated from the firmware type, it seems logical to me that `REPRAP_VOLUMETRIC` would cease to exist, because it would just be `REPRAP` with  `"machine_firmware_retract": false` in the .def.json.